crates/bevy_ui/src/layout/mod.rs

```diff
@@ -383,6 +383,34 @@ pub fn ui_layout_system(
     }
 }
 
+/// Resolve and update the widths of Node outlines
+pub fn resolve_outlines_system(
+    primary_window: Query<&Window, With<PrimaryWindow>>,
+    ui_scale: Res<UiScale>,
+    mut outlines_query: Query<(&Outline, &mut Node)>,
+) {
+    let viewport_size = primary_window
+        .get_single()
+        .map(|window| Vec2::new(window.resolution.width(), window.resolution.height()))
+        .unwrap_or(Vec2::ZERO)
+        / ui_scale.0 as f32;
+
+    for (outline, mut node) in outlines_query.iter_mut() {
+        let node = node.bypass_change_detection();
```

Comment:
How come you're calling `bypass_change_detection` here?

Reply:
The text system queries for changes to `Node` so it can recompute the text layout on changes to the UI node's size.

Reply:
So, do we not care about detecting changes to the `outline_width` and `outline_offset` fields?

Reply:
Not at the moment. Maybe we could have a separate component for every property but it doesn't seem very tractable.

Reply:
Fair enough. I'd put a comment here describing that we don't care about change detection for these fields because they're private and only used as a cache.